bin/picca_coadd_zint.py

```diff
@@ -0,0 +1,458 @@
+#!/usr/bin/env python3
+"""Coadd correlation function from different redshift intervals"""
+import os
+import argparse
+import fitsio
+import numpy as np
+
+from picca.utils import userprint
+
+
+def coadd_correlations(fi,fout):
+    """Coadds correlation functions measured in different redshift intervals.
+
+    Args:
+        fi: list of strings
+            List of the paths to the input correlations to be coadded.
+        fout: string
+            String of the path to the desired output.
+    """
+
+    # specify which header entries we want to check are consistent across all
+    # files being coadded
+    headers_to_check_match = ['NP','NT','OMEGAM','OMEGAR','OMEGAK','WL','NSIDE']
+
+    # initialize coadd arrays, fill them with zeros
+    with fitsio.FITS(fi[0]) as hdul:
+        r_par = hdul[1]['RP'][:] * 0
+        r_trans = hdul[1]['RT'][:] * 0
+        num_pairs = hdul[1]['NB'][:] * 0
+        z = hdul[1]['Z'][:] * 0
+        weights_total = r_par * 0
+        healpixs = hdul[2]['HEALPID'][:]
+
+        # get values of header entries to check with other files
+        header = hdul[1].read_header()
+        headers_to_check_match_values = {h:header[h] for h in headers_to_check_match}
+
+    # initialise header quantities
+    r_par_min = 1.e6
+    r_par_max = -1.e6
+    r_trans_max = -1.e6
+    z_cut_min = 1.e6
+    z_cut_max = -1.e6
+
+    # check to see if all files have the same HEALPix pixels
+    same_healpixs = True
+    for file in fi:
+        with fitsio.FITS(file) as hdul:
+            if len(hdul[2]['HEALPID'][:]) != len(healpixs):
+                same_healpixs *= False
+            elif (hdul[2]['HEALPID'][:] != healpixs).all():
+                same_healpixs *= False
+
+    # if so, initialise our xi and weights as arrays
+    if same_healpixs:
+        with fitsio.FITS(file) as hdul:
+            xi = hdul[2]['DA'][:] * 0
+            weights = hdul[2]['WE'][:] * 0
+    # otherwise, use dictionaries to match HEALPix pixels carefully (but slower)
+    else:
+        xi = {}
+        weights = {}
+
+    # loop over files
+    for file in fi:
+
+        # open the file and read the header
+        userprint("coadding file {}".format(file))
+        hdul = fitsio.FITS(file)
+        header = hdul[1].read_header()
+
+        # check that the header properties match those from the first file
+        for h in headers_to_check_match:
+            assert header[h] == headers_to_check_match_values[h]
+
+        # add weighted contributions from this file to coadd variables
+        weights_aux = hdul[2]['WE'][:]
```

Comment:
just to double check, `weights_aux` and `weights_total_aux` here are used to sum over all the different healpix, aren't they? If so, why does it not matter the value of `same_healpixs`?  Am I missing something here?

Reply:
Yes `weights_aux` is the weights in each separation bin and in each pixel, whereas `weights_total_aux` has been summed over pixels.

The value of `same_healpixs` doesn't matter as `weights_total_aux` is only used to provide appropriate weights for calculation the average redshift/separation etc of each separation bin, and we're only interested in the mean over all pixels in those cases. I.e. it doesn't matter if two correlations have measurements in completely different sets of pixels, we still would compute the mean z/r_par/r_trans in each bin over all pixels. Does that make sense? 

At any rate, using `same_healpixs` doesn't change any of the quantities being computed relative to the old method, it just makes things much much faster when our correlations have the exact same footprint

Reply:
Yes, this makes sense. I just wanted to make sure I was not missing anything